src/app/complex/complex-details/complex-references/complex-references.component.ts

```diff
@@ -1,91 +1,48 @@
-import {Component, Input, OnInit, input } from '@angular/core';
+import {Component, effect, input} from '@angular/core';
 import {CrossReference} from '../../shared/model/complex-details/cross-reference.model';
 
 @Component({
   selector: 'cp-complex-references',
   templateUrl: './complex-references.component.html',
   styleUrls: ['./complex-references.component.scss']
 })
-export class ComplexReferencesComponent implements OnInit {
-  private _crossReferences: CrossReference[];
-  private _pubmedXRefs: CrossReference[];
-  private _externalResources: CrossReference[];
-  private _synonyms: string[];
-  private _systematicName: string;
+export class ComplexReferencesComponent {
 
-  constructor() {
-  }
+  crossReferences = input<CrossReference[]>();
+  synonyms = input<string[]>();
+  systematicName = input<string>();
+  institution = input<string>();
+
+  pubmedXRefs: CrossReference[];
+  externalResources: CrossReference[];
 
-  ngOnInit() {
-    this.findXRefs();
+  constructor() {
+    effect(() => this.findXRefs(this.crossReferences()));
   }
 
   get hasXrefsOrSynonymsOrSystematicName(): boolean {
     return (this.pubmedXRefs && this.pubmedXRefs.length > 0) ||
-      (this.synonyms && this.synonyms.length > 0) ||
-      (!!this.systematicName && this.systematicName.length > 0);
+      (this.synonyms() && this.synonyms().length > 0) ||
+      (!!this.systematicName() && this.systematicName().length > 0);
   }
 
-  private findXRefs() {
-    for (let i = 0; i < this._crossReferences.length; i++) {
-      const crossRef = this._crossReferences[i];
-      const database = this._crossReferences[i].database.toLowerCase();
+  private findXRefs(crossReferences: CrossReference[]) {
```

Comment:
Will duplicate entries if new inputs are provided since you don't reinitialize fields upon new firing of this function. 
Ideally, to avoid this issue and make the code more declarative, I would refactor this a bit to use computed() instead of effect() in this case, since we basically just do a filter on the input